From 73f592a68c4a0cedcdf910c51c51a3359755c0e9 Mon Sep 17 00:00:00 2001
From: gimmy <gheorghe.popescu@puppet.com>
Date: Wed, 20 May 2020 12:44:50 +0300
Subject: [PATCH] force SONAME option for solaris
# Solaris machines need to provide the shared object name via the -h flag to the
# linker when compiling things. This is so when other things are linked against
# the lib just the lib name is listed (and not an entire path). See
# https://docs.oracle.com/cd/E19253-01/817-1984/chapter4-97194/index.html for a
# better description.

# This patch to the gcc jamfile forces the usage
# of the -h flag via SONAME_OPTION.


---
 tools/build/src/tools/gcc.jam | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/build/src/tools/gcc.jam b/tools/build/src/tools/gcc.jam
index 8910a55f2..b5bec7066 100644
--- a/tools/build/src/tools/gcc.jam
+++ b/tools/build/src/tools/gcc.jam
@@ -1070,6 +1070,9 @@ toolset.flags gcc.link.dll .IMPLIB-COMMAND <target-os>cygwin : "-Wl,--out-implib
     # See note [1]
     toolset.flags gcc.link OPTIONS <target-os>solaris/<runtime-link>static : -static ;

+    toolset.flags gcc.link HAVE_SONAME <target-os>solaris : "" ;
+    toolset.flags gcc.link SONAME_OPTION <target-os>solaris : -h ;
+
     # [1]
     # For <runtime-link>static we made sure there are no dynamic libraries in the
     # link. On HP-UX not all system libraries exist as archived libraries (for
--
2.25.0
